Update sssd_ldap_start_tls OVAL, bash and ansible remediations.

ggbecker opened the pull request:
#### Description:

- Update `sssd_ldap_start_tls` to align with latest version of RHEL7 STIG, and also fix a lot of minor issues with its content.

#### Rationale:

- Reference: https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-72227?version=v2r7

mildas commented:
**Changes identified:**
Rule sssd_ldap_start_tls:
 Node deleted from OVAL check.
 Node moved within OVAL check.
 Text added outsite tags in OVAL check.
 Attribute value changed in OVAL check.
 Ansible remediation changed.
 New node inserted to OVAL check.
 Found change in bash remediation.
 Text changed in OVAL check.
Profile stig on rhel7:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

redhatrises commented on linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_start_tls/rule.yml:
```diff
@@ -10,10 +10,10 @@ description: |-
-    If <tt>USELDAPAUTH=yes</tt>, then LDAP is being used. To check if LDAP is
-    configured to use TLS, use the following command:
+    If the <tt>sssd.service</tt> is <tt>active</tt>, then LDAP is being used.
+    To check if LDAP is configured to use TLS, use the following command:
```
Let's fix the rule description here as well, because sssd running doesn't mean that LDAP is being used.

````
    The LDAP client should be configured to implement TLS for the integrity
    of all remote LDAP authentication sessions. If the <tt>id_provider</tt> is
    set to <tt>ldap</tt> or <tt>ipa</tt> in <tt>/etc/sssd/sssd.conf</tt> or any of the
    <tt>/etc/sssd/sssd.conf.d</tt> configuration files, <tt>ldap_id_use_start_tls</tt>
    must be set to <tt>True</tt>.
    <br /><br />
    To check if LDAP is configured to use TLS when <tt>id_provider</tt> is
    set to <tt>ldap</tt> or <tt>ipa</tt>, use the following command:
    <pre>$ sudo grep -i ldap_id_use_start_tls /etc/sssd/sssd.conf</pre>
````

ggbecker replied:
Updated in 87c8ac087b45572760833e82f1197f019f7b101d

mildas commented:
**Changes identified:**
Rule sssd_ldap_start_tls:
 Node deleted from OVAL check.
 Node moved within OVAL check.
 Text added outsite tags in OVAL check.
 Found change in bash remediation.
 New node inserted to OVAL check.
 Text changed in OVAL check.
 Ansible remediation changed.
 Attribute value changed in OVAL check.
Profile stig on rhel7:
 Rule  added to stig profile.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls

mildas commented:
**Changes identified:**
Rule sssd_ldap_start_tls:
 New node inserted to OVAL check.
 Ansible remediation changed.
 Node moved within OVAL check.
 Node deleted from OVAL check.
 Attribute value changed in OVAL check.
 Deleted attribute from OVAL check.
 Found change in bash remediation.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
Rule checks:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule  added to stig profile.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

ggbecker commented:
I have added a new CPE (`sssd-ldap`) that detects that the sssd configuration is not using `ad` in `id_provider` parameter. And the idea is to mark rules that are only applicable when sssd is using `ldap` with this CPE. I think it aligns better with the text provided in the STIG which says that if sssd is not configured to use LDAP (it translates to something different than `ad`) then the item is not applicable.

Reference: https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-72227?version=v2r7

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Templatization usage changed.
 Found change in bash remediation.
Rule sssd_ldap_start_tls:
 Attribute value changed in OVAL check.
 Deleted attribute from OVAL check.
 Text added outsite tags in OVAL check.
 Node deleted from OVAL check.
 Templatization usage changed.
 New node inserted to OVAL check.
 Text changed in OVAL check.
 Found change in bash remediation.
 Node moved within OVAL check.
Rule checks:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_start_tls.
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
 In Bash remediation for sssd_ldap_start_tls.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Found change in bash remediation.
 Templatization usage changed.
Rule sssd_ldap_start_tls:
 Text changed in OVAL check.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Attribute value changed in OVAL check.
 Node moved within OVAL check.
 Found change in bash remediation.
 New node inserted to OVAL check.
 Templatization usage changed.
Rule checks:
 OVAL check is newly added.
 The rule doesn't occur in any profile nor product.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_start_tls.
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Found change in bash remediation.
 Templatization usage changed.
Rule sssd_ldap_start_tls:
 Text changed in OVAL check.
 Deleted attribute from OVAL check.
 Found change in bash remediation.
 Attribute value changed in OVAL check.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Text added outsite tags in OVAL check.
 Templatization usage changed.
 Node deleted from OVAL check.
Rule checks:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_start_tls.
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 (cd build && cmake ../ && ctest -j4)

jan-cerny commented on linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_start_tls/tests/ad_id_provider_and_tls_false.notapplicable.sh:
```diff
@@ -0,0 +1,7 @@
+# profiles = xccdf_org.ssgproject.content_profile_stig
+
+. $SHARED/setup_config_files.sh
+setup_correct_sssd_config
```
I think it should also ensure if sssd is installed.

ggbecker replied:
ok, I will take a look, from others test scenarios this should be enough:

```
yum -y install /usr/lib/systemd/system/sssd.service
systemctl enable sssd
```

jan-cerny replied:
That's amazing! It helps people like me who use the minimal RHEL VM for testing, as sssd isn't by default.

> yum -y install /usr/lib/systemd/system/sssd.service
> systemctl enable sssd

^ Probably vice-versa :smile:

ggbecker replied:
What do you mean by vice-versa here? Do you mean this: 03fe81459f0628bb08169086285cdf1cfef64e5b ?

jan-cerny reviewed (commented):
looks fine

ggbecker commented on linux_os/guide/services/sssd/sssd-ldap/sssd_ldap_start_tls/rule.yml:
```diff
@@ -5,15 +5,14 @@ prodtype: ol7,ol8,rhel7,rhel8,wrlinux1019
+    of all remote LDAP authentication sessions. If the <tt>id_provider</tt> is
+    set to <tt>ldap</tt> or <tt>ipa</tt> in <tt>/etc/sssd/sssd.conf</tt> or any of the
+    <tt>/etc/sssd/sssd.conf.d</tt> configuration files, <tt>ldap_id_use_start_tls</tt>
+    must be set to <tt>True</tt>.
```
I think it is case insensitive... but I'll change it to keep it concise

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Found change in bash remediation.
 Templatization usage changed.
Rule sssd_ldap_start_tls:
 Found change in bash remediation.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Attribute value changed in OVAL check.
 Node moved within OVAL check.
 Templatization usage changed.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
Rule checks:
 OVAL check is newly added.
 The rule doesn't occur in any profile nor product.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_start_tls.
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
 In Bash remediation for sssd_ldap_start_tls.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

matejak commented on shared/macros-oval.jinja:
```diff
@@ -534,3 +534,27 @@
+        elif grep -qs $DOMAIN_REGEX $SSSD_CONF; then
+                sed -i "/$DOMAIN_REGEX/a {{{ parameter }}} = {{{ value }}}" $SSSD_CONF
+        else
+                mkdir -p /etc/sssd
```
This has been touched in #5997 - if a basic directory that should be owned by a package is not present, then it is probably better to assume that the remediation is premature, i.e. not applicable.

jan-cerny replied:
+1

ggbecker replied:
I have removed the part of remediation that creates the file if doesn't exist similarly as it is done in #5997

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Found change in bash remediation.
 Templatization usage changed.
Rule sssd_ldap_start_tls:
 Text added outsite tags in OVAL check.
 Node deleted from OVAL check.
 Text changed in OVAL check.
 Templatization usage changed.
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Found change in bash remediation.
 Deleted attribute from OVAL check.
 New node inserted to OVAL check.
Rule checks:
 OVAL check is newly added.
 The rule doesn't occur in any profile nor product.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
 In Ansible remediation for sssd_ldap_start_tls.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig
 (cd build && cmake ../ && ctest -j4)

jan-cerny commented:
@openscap-ci test this please

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Templatization usage changed.
 Found change in bash remediation.
Rule sssd_ldap_start_tls:
 Node moved within OVAL check.
 Node deleted from OVAL check.
 Templatization usage changed.
 Text changed in OVAL check.
 Text added outsite tags in OVAL check.
 Found change in bash remediation.
 New node inserted to OVAL check.
 Deleted attribute from OVAL check.
 Attribute value changed in OVAL check.
Rule checks:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_start_tls.
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig
 (cd build && cmake ../ && ctest -j4)
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls

jan-cerny commented:
Hi @ggbecker , I'm getting the following error on RHEL 7:

```
ERROR - Script ldap_id_provider_and_tls_false.fail.sh using profile xccdf_org.ssgproject.content_profile_stig found issue:
ERROR - Rule evaluation resulted in pass, instead of expected fail during initial stage 
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_sssd_ldap_start_tls'.
```

Can you check it?

ggbecker commented:
> Hi @ggbecker , I'm getting the following error on RHEL 7:
> 
> ```
> ERROR - Script ldap_id_provider_and_tls_false.fail.sh using profile xccdf_org.ssgproject.content_profile_stig found issue:
> ERROR - Rule evaluation resulted in pass, instead of expected fail during initial stage 
> ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_sssd_ldap_start_tls'.
> ```
> 
> Can you check it?

I have fixed the test scenarios... since we changed from `True` to `true` in test scenarios, it was needed to use `sed` with case insensitive option to replace values from the sample `sssd.conf` we use to test. It should be working now. I'm gonna force push the fix.

mildas commented:
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Templatization usage changed.
 Found change in bash remediation.
Rule sssd_ldap_start_tls:
 New node inserted to OVAL check.
 Node deleted from OVAL check.
 Text added outsite tags in OVAL check.
 Attribute value changed in OVAL check.
 Text changed in OVAL check.
 Found change in bash remediation.
 Templatization usage changed.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
Rule checks:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.
Profile stig on rhel7:
 Rule  added to stig profile.
Macro ansible_sssd_ldap_config:
 In Ansible remediation for sssd_ldap_configure_tls_ca_dir.
 In Ansible remediation for sssd_ldap_start_tls.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_start_tls.
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
Others:
 Python abstract syntax tree change found in ssg/constants.py.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 (cd build && cmake ../ && ctest -j4)
 build_product rhel7
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhel7-ds.xml stig

jan-cerny commented:
Great! It passes now.
